### PR TITLE
tests(dm): migrate pingcap/check to testify for pkg/binlog

### PR DESCRIPTION
**Pull request by liumengya94**
### What problem does this PR solve?

migrate pingcap/check to testify for pkg/binlog

Issue Number: ref #5687

### What is changed and how it works?
Use testify in dm/pkg/binlog package

#### Tests 

 - Unit test

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
None
```

### CODE REVIEW

**Review by lance6716 — APPROVED** [review]

File: dm/pkg/binlog/common/replication_test.go [hunk 1]
```diff
@@ -15,19 +15,20 @@ package common
 
 import (
 	"github.com/go-mysql-org/go-mysql/replication"
-	. "github.com/pingcap/check"
+	"github.com/stretchr/testify/require"
+	"testing"
```
Comment (gozssky) [inline dm/pkg/binlog/common/replication_test.go:19]: Please run `make fmt` to fix the import order.

**Review by gozssky — APPROVED** [review]

File: dm/pkg/binlog/event/common_test.go [hunk 2]
```diff
@@ -107,96 +106,97 @@ func (t *testCommonSuite) TestGenCommonGTIDEvent(c *C) {
 
 	// nil gSet, invalid
 	gtidEv, err := GenCommonGTIDEvent(flavor, serverID, latestPos, gSet, false, 0)
-	c.Assert(err, NotNil)
-	c.Assert(gtidEv, IsNil)
+	require.NotNil(t, err)
```
Comment (lance6716) [inline dm/pkg/binlog/event/common_test.go:109]: require.Error

**Comment by lance6716** [comment]
/hold

lint error

```

[2022-11-15T05:56:58.851Z] pkg/binlog/position_test.go:24:6: type `testPositionSuite` is unused (unused)

[2022-11-15T05:56:58.851Z] type testPositionSuite struct{}

[2022-11-15T05:56:58.851Z]      ^

[2022-11-15T05:56:58.851Z] pkg/binlog/pos_finder_test.go:328: unnecessary trailing newline (whitespace)

[2022-11-15T05:56:58.851Z] 

[2022-11-15T05:56:58.851Z] 	}

[2022-11-15T05:56:58.851Z] pkg/binlog/event/event_test.go:56:6: test helper function should start from t.Helper() (thelper)

[2022-11-15T05:56:58.851Z] func verifyHeader(t *testing.T, obtained, excepted *replication.EventHeader, eventType replication.EventType, latestPos, eventSize uint32) {

[2022-11-15T05:56:58.851Z]      ^

[2022-11-15T05:56:58.851Z] pkg/binlog/event/generator_test.go:122:6: test helper function should start from t.Helper() (thelper)

[2022-11-15T05:56:58.851Z] func testGenerate(t *testing.T, flavor string, serverID uint32, latestGTID gmysql.GTIDSet, previousGTIDSet gmysql.GTIDSet, latestXID uint64) {

[2022-11-15T05:56:58.851Z]      ^

[2022-11-15T05:56:58.851Z] pkg/binlog/event/generator_test.go:305:6: test helper function should start from t.Helper() (thelper)

[2022-11-15T05:56:58.851Z] func previousGTIDEventType(t *testing.T, flavor string) replication.EventType {

[2022-11-15T05:56:58.851Z]      ^

[2022-11-15T05:56:58.851Z] make: *** [check-static] Error 1
```

unit test failed

```
[2022-11-15T06:00:04.824Z] === Failed

[2022-11-15T06:00:04.824Z] === FAIL: dm/pkg/binlog/reader TestSyncPos (0.00s)

[2022-11-15T06:00:04.824Z] [2022/11/15 13:54:40] [info] binlogsyncer.go:164 create BinlogSyncer with config {3251 mysql  0     false false <nil> false UTC false 0 0s 0s 0 false false 0 <nil> 0xc0000534a0 0x91cf40}

[2022-11-15T06:00:04.824Z] [2022/11/15 13:54:40] [info] binlogsyncer.go:164 create BinlogSyncer with config {3252 mysql  0     false false <nil> false UTC false 0 0s 0s 0 false false 0 <nil> 0xc0005a6000 0x91cf40}

[2022-11-15T06:00:04.824Z]     tcp_test.go:69: 

[2022-11-15T06:00:04.824Z]         	Error Trace:	/home/jenkins/agent/workspace/ut-check/tiflow/dm/pkg/binlog/reader/tcp_test.go:69

[2022-11-15T06:00:04.824Z]         	Error:      	Expected nil, but got: [code=30012:class=relay-unit:scope=upstream:level=high], Message: start sync from position (, 0), RawCause: dial tcp: missing address

[2022-11-15T06:00:04.824Z]         	Test:       	TestSyncPos

[2022-11-15T06:00:04.824Z] 

[2022-11-15T06:00:04.824Z] === FAIL: dm/pkg/binlog/reader TestSyncGTID (0.00s)

[2022-11-15T06:00:04.824Z]     tcp_test.go:124: 

[2022-11-15T06:00:04.824Z]         	Error Trace:	/home/jenkins/agent/workspace/ut-check/tiflow/dm/pkg/binlog/reader/tcp_test.go:124

[2022-11-15T06:00:04.824Z]         	Error:      	Expected nil, but got: [code=30014:class=relay-unit:scope=upstream:level=high], Message: start sync from GTID set , RawCause: dial tcp: missing address

[2022-11-15T06:00:04.824Z]         	Test:       	TestSyncGTID
```

File: dm/pkg/binlog/reader/tcp_test.go [hunk 3]
```diff
@@ -21,130 +21,131 @@ import (
...
+	suite.Suite
+}
 
-func (t *testTCPReaderSuite) SetUpSuite(c *C) {
-	c.Assert(failpoint.Enable("github.com/pingcap/tiflow/dm/pkg/binlog/reader/MockTCPReaderStartSyncByPos", "return(true)"), IsNil)
-	c.Assert(failpoint.Enable("github.com/pingcap/tiflow/dm/pkg/binlog/reader/MockTCPReaderStartSyncByGTID", "return(true)"), IsNil)
-	c.Assert(failpoint.Enable("github.com/pingcap/tiflow/dm/pkg/binlog/reader/MockTCPReaderClose", "return(true)"), IsNil)
-	c.Assert(failpoint.Enable("github.com/pingcap/tiflow/dm/pkg/binlog/reader/MockTCPReaderGetEvent", "return(true)"), IsNil)
-	c.Assert(failpoint.Enable("github.com/pingcap/tiflow/dm/pkg/binlog/reader/MockTCPReaderStatus", "return(true)"), IsNil)
+func (t *testTCPReaderSuite) SetUpSuite() {
```
Comment (lance6716) [inline dm/pkg/binlog/reader/tcp_test.go:44]: should be `SetupSuite`

you can check the doc about this interface

https://pkg.go.dev/github.com/stretchr/testify/suite#SetupAllSuite

https://pkg.go.dev/github.com/stretchr/testify/suite#TearDownAllSuite

**Pull request merged**